### ISSUE TITLE
Gurubashi Axe Thrower was too powerful

### ISSUE THREAD

**Issue by yushicsm** [cmangos/classic-db#128]
"Throw Axe" sould attack one player once，but attack all players once now.
That cause all players dead quickly...
Maybe too powerful...

**Comment by Phatcat** [comment]
> "Throw Axe" sould attack one player once，but attack all players once now.

Are you sure about that? Are you not refering to 'Axe flurry'? Cause I don't see how 'Throw Axe' should be hitting more than one target? Axe flurry however, is indeed supposed to hit everyone in range;

> Id: 16075
Name: Throw Axe
ToolTip: Stunned.
Effects
Effect 0:
Id: 2 (SPELL_EFFECT_SCHOOL_DAMAGE)
BasePoints = 1750
Targets (6, 0) (TARGET_CHAIN_DAMAGE, NO_TARGET)
Effect 1:
Id: 6 (SPELL_EFFECT_APPLY_AURA)
BasePoints = 1
Targets (6, 0) (TARGET_CHAIN_DAMAGE, NO_TARGET)
Aura Id 12 (SPELL_AURA_MOD_STUN), value = 1, misc = 0 (0), periodic = 0

As far as I know, target (6, 0) is a single target target..

> Id: 24019
Name: Axe Flurry
Parent spell: 24018 - Axe Flurry
Effects
Effect 0:
Id: 3 (SPELL_EFFECT_DUMMY)
BasePoints = 1
Targets (22, 15) (TARGET_CASTER_COORDINATES, TARGET_ALL_ENEMY_IN_AREA)
Radius (Id 10) 30.00

>Id: 24071
Name: Axe Flurry
Parent spell: 24018 - Axe Flurry
Effects
Effect 0:
Id: 58 (SPELL_EFFECT_WEAPON_DAMAGE)
BasePoints = 0
Targets (22, 15) (TARGET_CASTER_COORDINATES, TARGET_ALL_ENEMY_IN_AREA)
Radius (Id 8) 5.00

As you can see, it's meant to hit everyone in a 30 yard range with ranged attacks as well as anyone in a 5 yard range with melee damage as well

**Comment by Phatcat** [comment]
Hmm..

> Id: 421
Name: Chain Lightning (Rank 1)
Description: Hurls a lightning bolt at the enemy, dealing $s1 Nature damage and then jumping to additional nearby enemies. Each jump reduces the damage by 30%. Affects $x1 total targets.
Parent spell: 920 - Chain Lightning (Rank 1)
Effects
Effect 0:
Id: 2 (SPELL_EFFECT_SCHOOL_DAMAGE)
BasePoints = 191 + Level * 1.90 to 218 + lvl * 1.90 * 0.70
Targets (6, 0) (TARGET_CHAIN_DAMAGE, NO_TARGET)
EffectChainTarget = 3

So target (6, 0) (TARGET_CHAIN_DAMAGE, NO_TARGET) is, or can at least potentially indeed be more than a single target... the abscence of a EffectChainTarget value from 'Throw Axe' however makes me think the axe don't/shouldn't jump to additional targets beyond the inial one... Does it currently?

**Comment by yushicsm** [comment]
what tools you use to see the spells....

**Comment by killerwife** [comment]
QSW - QT spellworks from sidsukana, google it. There is a github with releases.

**Comment by yushicsm** [comment]
thanks

**Comment by yushicsm** [comment]
I test again
16057 seems not have problem。
but 24019 must have problem，it should hit one player once，not hit all players once now.

**Comment by Phatcat** [comment]
> but 24019 must have problem，it should hit one player once，not hit all players once now.

No... It should hit all players... and not just once.. it's a channeled attack... Read the target of 24019;

> Targets (22, 15) (TARGET_CASTER_COORDINATES, TARGET_ALL_ENEMY_IN_AREA)

Notice it says 'ALL ENEMY IN AREA' - that means all players indeed, and their pets as well.

**Comment by cala** [comment]
It is possible that we have to force in core to only pick a given number of targets amongst TARGET_ALL_ENEMY_IN_AREA . We already to that for other spells. When I did a run in ZG two weeks ago, I noticed that this NPCs were way much harder than any other ZG or AQ20 trash but I can't say if their behaviour is not what it is supposed to be.

**Comment by Phatcat** [comment]
Indeed that would seem probable since it does have a dummy wrapper for the actual impact spell, the one implemented not too long ago. Surely some sort of logic other than simply casting a spell should go in there, I just wasn't sure what, and I tried doing a 50/50 chance on getting hit or not but I wasn't happy with the result since it wasn't dependent on number of players within range, so I just settled on simply casting the impact spell. I didn't know it was being done with others, cause that seems very plausible to also do here. question then becomes... how many targets? and should it be the closest targets, or random targets within range...
